app/assets/javascripts/discourse/tests/acceptance/composer-test.js

```diff
@@ -1106,6 +1106,55 @@ acceptance("Composer - Customizations", function (needs) {
   });
 });
 
+acceptance("Composer - Error Extensibility", function (needs) {
+  needs.user();
+  needs.settings({
+    general_category_id: 1,
+    default_composer_category: 1,
+  });
+
+  needs.hooks.beforeEach(() => {
+    withPluginApi("1.5.0", (api) => {
+      api.addComposerSaveErrorCallback((error) => {
+        if (error.match(/PLUGIN_XYZ ERROR/)) {
+          // handle error
+          return true;
+        }
+        return false;
+      });
+    });
+  });
+
+  test("Create a topic with server side errors handled by a plugin", async function (assert) {
+    pretender.post("/posts", function () {
+      return response(422, { errors: ["PLUGIN_XYZ ERROR"] });
+    });
+
+    await visit("/");
+    await click("#create-topic");
+    await fillIn("#reply-title", "this title triggers an error");
+    await fillIn(".d-editor-input", "this is the *content* of a post");
+    await click("#reply-control button.create");
+    assert.notOk(exists(".dialog-body"), "it does not pop up an error message");
+  });
+
+  test("Create a topic with server side errors not handled by a plugin", async function (assert) {
+    pretender.post("/posts", function () {
+      return response(422, { errors: ["PLUGIN_ABC ERROR"] });
+    });
+
+    await visit("/");
+    await click("#create-topic");
+    await fillIn("#reply-title", "this title triggers an error");
+    await fillIn(".d-editor-input", "this is the *content* of a post");
+    await click("#reply-control button.create");
+    assert.ok(exists(".dialog-body"), "it pops up an error message");
+    await click(".dialog-footer .btn-primary");
+    assert.ok(!exists(".dialog-body"), "it dismisses the error");
+    assert.ok(exists(".d-editor-input"), "the composer input is visible");
```

Comment:
Should we check that `PLUGIN_AGC` message is included in the dialog text here?

Reply:
Handled in 7cd47d84c5